RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -999,8 +999,8 @@ void CaloTowersCreationAlgo::convert(const CaloTowerDetId& id, const MetaTower&
     
     
     // insert in collection (remove and return if below threshold)
-    if unlikely ( (towerP4[3]==0) & (E_outer>0)  ) {
-      collection.emplace_back(id, E_em, E_had, E_outer, -1, -1, CaloTower::PolarLorentzVector(0,hadPoint.eta(), hadPoint.phi(),0),  emPoint, hadPoint);
+    if unlikely ( (towerP4[3]==0) & (E_outer>0) & (!theHOIsUsed) ) {
+      collection.emplace_back(id, E_em, E_had, E_outer, -1, -1, CaloTower::PolarLorentzVector(1e-9,hadPoint.eta(), hadPoint.phi(),0),  emPoint, hadPoint);
```

Comment:
sorry, but this will change behavior for theHOIsUsed=true (I didn't inspect the code carefully enough to see if it has a valid use case. If I understand correctly, your issue is only for theHOIsUsed=false.
```
if unlikely ( (towerP4[3]==0) & (E_outer>0)  ) {
  float val = theHOIsUsed ? 0 : 1E-9;
  collection.emplace_back(id, E_em, E_had, E_outer, -1, -1, CaloTower::PolarLorentzVector(val,hadPoint.eta(), hadPoint.phi(),0),  emPoint, hadPoint); 
```